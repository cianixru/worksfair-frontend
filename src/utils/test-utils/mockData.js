// eslint-disable-next-line import/prefer-default-export
export const user = {
  username: 'Theookafor2',
  first_name: 'eod',
  last_name: 'okafor',
  email: 'theodorehope@andela.com',
  facebook: null,
  instagram: null,
  twitter: null,
  headline: null,
  current_location: null,
  role: 'admin',
  confirmed_account: false,
  verified_account: false,
  reset_token: null,
  image_url: null,
  created_at: '2019-03-10T08:48:42.942532Z',
  modified_at: '2019-03-10T08:48:42.942544Z',
  token: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9',
};

export const webpage = {
  owner: {
    id: 1,
    username: 'theowa',
    first_name: 'theo',
    last_name: 'IO',
    email: 'theo.iowa@yahoo.com',
    facebook: null,
    instagram: null,
    twitter: null,
    headline: null,
    current_location: null,
    role: 'admin',
    confirmed_account: false,
    verified_account: false,
    reset_token: null,
    image_url: 'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png',
    created_at: '2019-04-25T09:07:08.209916Z',
    modified_at: '2019-04-25T09:07:08.209930Z'
  },
  title: 'Happiness',
  sub_domain_name: 'happiness',
  description: 'Happiness truly lives here. We just happy',
  keywords: 'passion, software, engineering, technology, worksfair',
  country: 'Nigeria',
  state: 'Lagos',
  city: 'Okota',
  area: 'Omagba Phase 1, Phase 2, Abatete',
  address: 'No 7 Obinna Uzor Street, Canal Estate',
  phone: null,
  email: null,
  website: null,
  featured_images: [
    'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png',
    'https://res.cloudinary.com/lbgcghcovyvtdbpih8ba.png'
  ],
  is_active: true,
  logo: null,
  colour: 'is-dark',
  created_at: '2019-04-28T18:39:25.174272Z',
  updated_at: '2019-05-01T12:18:49.669633Z'
};
