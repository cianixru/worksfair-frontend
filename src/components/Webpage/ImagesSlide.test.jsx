import React from 'react';
import { render } from 'react-testing-library';
import { BrowserRouter as Router } from 'react-router-dom';

import ImageSlide from './ImagesSlide';
import { webpage } from '../../utils/test-utils/mockData';

test('Test that the ImageSlide displays default pictures', () => {
  webpage.featured_images = null;
  const component = render(
    <Router>
      <ImageSlide webpage={webpage} />
    </Router>
  );
  expect(component).toMatchSnapshot();
});
