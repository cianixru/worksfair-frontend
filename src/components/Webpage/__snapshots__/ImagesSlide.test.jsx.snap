// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that the ImageSlide displays default pictures 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="carousel carousel-slider"
          style="width: 100%;"
        >
          <button
            class="control-arrow control-prev"
            type="button"
          />
          <div
            class="slider-wrapper axis-horizontal"
          >
            <ul
              class="slider animated"
              style="transform: translate3d(-50%,0,0); transition-duration: 350ms;"
            >
              <li
                class="slide"
                style="min-width: 40%;"
              >
                <figure
                  class="image is-4by3"
                >
                  <img
                    alt="sample"
                    src="sample-04.jpg"
                  />
                </figure>
              </li>
              <li
                class="slide"
                style="min-width: 40%;"
              >
                <figure
                  class="image is-4by3"
                >
                  <img
                    alt="sample"
                    src="sample-01.jpg"
                  />
                </figure>
              </li>
              <li
                class="slide selected"
                style="min-width: 40%;"
              >
                <figure
                  class="image is-4by3"
                >
                  <img
                    alt="sample"
                    src="sample-02.jpg"
                  />
                </figure>
              </li>
              <li
                class="slide"
                style="min-width: 40%;"
              >
                <figure
                  class="image is-4by3"
                >
                  <img
                    alt="sample"
                    src="sample-03.jpg"
                  />
                </figure>
              </li>
              <li
                class="slide"
                style="min-width: 40%;"
              >
                <figure
                  class="image is-4by3"
                >
                  <img
                    alt="sample"
                    src="sample-04.jpg"
                  />
                </figure>
              </li>
              <li
                class="slide"
                style="min-width: 40%;"
              >
                <figure
                  class="image is-4by3"
                >
                  <img
                    alt="sample"
                    src="sample-01.jpg"
                  />
                </figure>
              </li>
            </ul>
          </div>
          <button
            class="control-arrow control-next"
            type="button"
          />
          <ul
            class="control-dots"
          >
            <li
              class="dot"
              role="button"
              tabindex="0"
              value="0"
            />
            <li
              class="dot selected"
              role="button"
              tabindex="0"
              value="1"
            />
            <li
              class="dot"
              role="button"
              tabindex="0"
              value="2"
            />
            <li
              class="dot"
              role="button"
              tabindex="0"
              value="3"
            />
          </ul>
          <p
            class="carousel-status"
          >
            2 of 4
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="carousel carousel-slider"
        style="width: 100%;"
      >
        <button
          class="control-arrow control-prev"
          type="button"
        />
        <div
          class="slider-wrapper axis-horizontal"
        >
          <ul
            class="slider animated"
            style="transform: translate3d(-50%,0,0); transition-duration: 350ms;"
          >
            <li
              class="slide"
              style="min-width: 40%;"
            >
              <figure
                class="image is-4by3"
              >
                <img
                  alt="sample"
                  src="sample-04.jpg"
                />
              </figure>
            </li>
            <li
              class="slide"
              style="min-width: 40%;"
            >
              <figure
                class="image is-4by3"
              >
                <img
                  alt="sample"
                  src="sample-01.jpg"
                />
              </figure>
            </li>
            <li
              class="slide selected"
              style="min-width: 40%;"
            >
              <figure
                class="image is-4by3"
              >
                <img
                  alt="sample"
                  src="sample-02.jpg"
                />
              </figure>
            </li>
            <li
              class="slide"
              style="min-width: 40%;"
            >
              <figure
                class="image is-4by3"
              >
                <img
                  alt="sample"
                  src="sample-03.jpg"
                />
              </figure>
            </li>
            <li
              class="slide"
              style="min-width: 40%;"
            >
              <figure
                class="image is-4by3"
              >
                <img
                  alt="sample"
                  src="sample-04.jpg"
                />
              </figure>
            </li>
            <li
              class="slide"
              style="min-width: 40%;"
            >
              <figure
                class="image is-4by3"
              >
                <img
                  alt="sample"
                  src="sample-01.jpg"
                />
              </figure>
            </li>
          </ul>
        </div>
        <button
          class="control-arrow control-next"
          type="button"
        />
        <ul
          class="control-dots"
        >
          <li
            class="dot"
            role="button"
            tabindex="0"
            value="0"
          />
          <li
            class="dot selected"
            role="button"
            tabindex="0"
            value="1"
          />
          <li
            class="dot"
            role="button"
            tabindex="0"
            value="2"
          />
          <li
            class="dot"
            role="button"
            tabindex="0"
            value="3"
          />
        </ul>
        <p
          class="carousel-status"
        >
          2 of 4
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
